Sources/MediaScrubProgressBar.swift

```diff
@@ -0,0 +1,167 @@
+        label.setContentHuggingPriority(.defaultHigh, for: .vertical)
+        label.textAlignment = .center
+        return label
+    }()
+
+    private lazy var scrubLabelStackView: UIStackView = {
+        let scrubLabelStack = UIStackView(arrangedSubviews: [scrubbingIndicatorLabel, scrubbingHelpLabel])
+        scrubLabelStack.axis = .vertical
+        scrubLabelStack.isHidden = true
+        scrubLabelStack.backgroundColor = UIColor(0x000000, 0.6)
+        return scrubLabelStack
+    }()
+    
+    // MARK: Initializers
+    required init(coder: NSCoder) {
+        fatalError("init(coder: NSCoder) not implemented")
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupViews()
+    }
+    
+    // MARK: Instance Methods
+    private func setupViews() {
+        let horizontalStack = UIStackView(arrangedSubviews: [elapsedTimeLabel, remainingTimeLabel])
+        horizontalStack.distribution = .equalSpacing
+        addArrangedSubview(scrubLabelStackView)
+        addArrangedSubview(horizontalStack)
+        addArrangedSubview(progressSlider)
+        spacing = 5
+        axis = .vertical
+        translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    private func updateScrubLabel() {
+        switch progressSlider.scrubbingSpeed {
+        case 1.0:
+            scrubbingIndicatorLabel.text = NSLocalizedString("PLAYBACK_SCRUB_HIGH", comment:"")
+        case 0.5:
+            scrubbingIndicatorLabel.text = NSLocalizedString("PLAYBACK_SCRUB_HALF", comment: "")
+        case 0.25:
+            scrubbingIndicatorLabel.text = NSLocalizedString("PLAYBACK_SCRUB_QUARTER", comment: "")
+        default:
+            scrubbingIndicatorLabel.text = NSLocalizedString("PLAYBACK_SCRUB_FINE", comment: "")
+        }
+    }
+
+    @objc private func updatePlaybackPosition() {
+        if !positionSet {
+            playbackController.playbackPosition = progressSlider.value
+            playbackController.setNeedsMetadataUpdate()
+            positionSet = true
+        }
+    }
+
+    @objc func updateForScrubbing() {
+        if !isScrubbing {
+            progressSlider.value = playbackController.playbackPosition
+        }
+        remainingTimeLabel.text = playbackController.remainingTime().stringValue
+        elapsedTimeLabel.text = playbackController.playedTime().stringValue
+    }
+    
+    // MARK: Slider Methods
+    @objc private func moveSliderThumb() {
+        perform(#selector(updatePlaybackPosition), with: nil, afterDelay: 0.3)
+        if playbackController.mediaDuration > 0 {
+            updateForScrubbing()
+        }
+        positionSet = false
+        delegate?.mediaSrubProgressBarShouldResetIdleTimer()
+    }
+
+    @objc private func progressSliderTouchDown() {
+        updateScrubLabel()
+        isScrubbing = true
+        scrubLabelStackView.isHidden = !isScrubbing
+        delegate?.mediaProgressScrubBarDidUpdateScrubbing(isScrubbing: isScrubbing)
```

Comment:
can we have a did start scrubbing here?

Reply:
but wouldnt that be creating more biolerplate code, because I'd have two functions that basically just set the scrubActivity indicator to hidden or visible